src/video/sdl2_v.cpp

```diff
@@ -664,19 +701,104 @@ void VideoDriver_SDL::Stop()
 	}
 }
 
-void VideoDriver_SDL::MainLoop()
+void VideoDriver_SDL::LoopOnce()
 {
-	uint32 cur_ticks = SDL_GetTicks();
-	uint32 last_cur_ticks = cur_ticks;
-	uint32 next_tick = cur_ticks + MILLISECONDS_PER_TICK;
 	uint32 mod;
 	int numkeys;
 	const Uint8 *keys;
+	uint32 prev_cur_ticks = cur_ticks; // to check for wrapping
+	InteractiveRandom(); // randomness
+
+	while (PollEvent() == -1) {}
+	if (_exit_game) {
+#ifdef __EMSCRIPTEN__
+		emscripten_cancel_main_loop();
+		MainLoopCleanup();
```

Comment:
Previously this was called when the main loop was broken out of here, now it is not

Reply:
This is when the mainloop is broken out of. Emscripten unrolls the main loop, and as such, it no longer exists. When `_exit_game` goes true, it normally breaks out of the mainloop, calling this function. And now we do it here :) Will leave a comment!